GH-10083: Replace springframework Nullable with jspecify Nullable in spring-integration-camel

anayonkars opened the pull request:
This change is applicable only to `spring-integration-camel`. 
Replaced `org.springframework.lang.Nullable` with `org.jspecify.annotations.Nullable`.

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

anayonkars committed:
Issue 10083: Apply Nullability to the project

This change is applicable only to spring-integration-camel.
Replaced org.springframework.lang.Nullable with org.jspecify.annotations.Nullable.

Signed-off-by: Anayonkar Shivalkar <anayonkar.shivalkar@gmail.com>

artembilan reviewed (changes requested):
Same here, @anayonkars ,
Please, migrate `package-info.java` in this module to the `@org.jspecify.annotations.NullMarked` and run `gradlew :spring-integration:check` locally.

artembilan reviewed (changes requested):
I mean that is essentially what `Apply Nullability` is about.
And, please, be more specific in the title of your PR (commit headline, to be precise).
I mean you don't apply the change for the whole project, but rather specific module, so would be great to reflect that in the PR title.

Let me know how that `@NullMarked` goes!

Thanks

anayonkars committed:
Issue 10083: Apply Nullability to the project

This change is applicable only to spring-integration-camel.
Handled changes in package-info.java.
Handled nullable related issues in CamelMessageHandler.

Signed-off-by: Anayonkar Shivalkar <anayonkar.shivalkar@gmail.com>

anayonkars commented:
@artembilan I've made changes to PR's title.
Also made relevant changes in `package-info.java`.
Had to do few more changes in `CamelMessageHandler` due to nullable related issues.
Kindly review.

artembilan commented on spring-integration-camel/src/main/java/org/springframework/integration/camel/outbound/CamelMessageHandler.java:
```diff
@@ -83,6 +85,7 @@ public class CamelMessageHandler extends AbstractReplyProducingMessageHandler {
 	@Nullable
 	private Expression exchangePropertiesExpression;
 
+	@Nullable
```
This has to be mitigated as `@SuppressWarnings("NullAway.Init")`.
It is initialized from the `afterPropertiesSet()`.
See more info in docs: https://docs.spring.io/spring-framework/reference/7.0/core/null-safety.html#_warnings_suppression

anayonkars replied:
Thanks! I've handled this as per review comments.

artembilan replied:
Thanks, but you don't have to say this since I got GH notification about a new commit in this PR.
I believe if you agreed with me than I would see the requested change in that commit.
If you don't agree, then feel free to comment and start discussion about the possible path.

Sometime I have like one hundred notifications about updates in my projects and simple `Done` in my mail box brings extra unnecessary noise 😄

artembilan commented on spring-integration-camel/src/main/java/org/springframework/integration/camel/outbound/CamelMessageHandler.java:
```diff
@@ -197,46 +200,57 @@ public void configure() throws Exception {
+	@Nullable
 	protected Object handleRequestMessage(Message<?> requestMessage) {
+		StandardEvaluationContext localEvaluationContext = Objects.requireNonNull(this.evaluationContext,
+				"'evaluationContext' must not be null");
```
With the `NullAway.Init` we don't need this anymore.

artembilan commented on spring-integration-camel/src/main/java/org/springframework/integration/camel/outbound/CamelMessageHandler.java:
```diff
@@ -197,46 +200,57 @@ public void configure() throws Exception {
 		Endpoint endpoint = resolveEndpoint(requestMessage);
 		Exchange exchange = prepareInExchange(endpoint, exchangePattern, requestMessage);
 
+		ProducerTemplate localProducerTemplate = Objects.requireNonNull(this.producerTemplate,
```
Same with `this.producerTemplate`

anayonkars committed:
GH-10083: Replace springframework Nullable with jspecify Nullable in spring-integration-camel

This change is applicable only to spring-integration-camel.
Fixed issues in CamelMessageHandler as per review comments (suppressed warnings for NullAway.Init so that Objects.requireNonNull is not needed).

Signed-off-by: Anayonkar Shivalkar <anayonkar.shivalkar@gmail.com>
